actors/util/adt/balancetable.go

```diff
@@ -73,6 +94,23 @@ func (t *BalanceTable) SubtractWithMininum(key addr.Address, req abi.TokenAmount
 	return sub, nil
 }
 
+// Subtracts up to the specified amount from a balance, without reducing the balance below zero.
+// Returns the amount subtracted (always positive or zero).
+func (t *BalanceTable) SubtractNonZero(key addr.Address, req abi.TokenAmount) (abi.TokenAmount, error) {
```

Comment:
FWIW I think you should just use `t.SubtractWithMinimum(key, req, big.NewInt(0))` at the call site. I already did that in the power actor.